SerialPrograms/Source/ML/Inference/ML_YOLOv5Detector.cpp

```diff
@@ -89,7 +91,31 @@ bool YOLOv5Detector::detect(const ImageViewRGB32& screen){
     cv::cvtColor(frame_mat_bgra, frame_mat_rgb, cv::COLOR_BGRA2RGB);
     
     m_output_boxes.clear();
-    m_yolo_session->run(frame_mat_rgb, m_output_boxes);
+
+    // fall back to CPU if fails with GPU.
+    for(size_t i = 0;;i++){

```

Comment:
Same here. Don't use for infinite for loop when you can just have two calls of the session run, one for gpu, and one for cpu if exception is caught.